region/lowernorfair/east/Fast Pillars Setup Room.json

```diff
@@ -496,6 +496,441 @@
+      "requires": [
+        "canCarefulJump",
+        {"heatFrames": 90}
+      ],
+      "note": [
+        "Come in with blue speed, killing the lower two Pirates, with enough momentum to jump all the way up to the door.",
+        "Then either quickly jump through the door before the top Pirate fires its laser, or wait for it to jump to the other side and then jump over it."
+      ],
+      "devNote": [
+        "This can also be done with lower run speeds in specific ranges, namely between $2.0 and $2.1 or between $2.E and $3.1, but this would be much more difficult."
+      ]
+    },
+    {
+      "link": [1, 4],
+      "name": "Come In Getting Blue Speed, Speedy Jump",
+      "entranceCondition": {
+        "comeInGettingBlueSpeed": {
+          "length": 8,
+          "openEnd": 1,
+          "minExtraRunSpeed": "$3.E"
+        }
+      },
+      "requires": [
+        {"or": [
+          "canDodgeWhileShooting",
+          {"enemyDamage": {
+            "enemy": "Yellow Space Pirate (standing)",
+            "type": "contact",
+            "hits": 1
+          }}
+        ]},
+        {"heatFrames": 135}
+      ],
+      "note": [
+        "Come in with blue speed, with enough momentum to jump, kill the lower two Pirates, and land on the platform above.",
+        "Then either quickly jump through the door before the top Pirate fires its laser, or wait for it to jump to the other side and then jump over it."
+      ],
+      "devNote": [
+        "This can also be done with lower run speeds in specific ranges, namely between $2.0 and $2.1 or between $2.E and $3.1, but this would be much more difficult."
+      ]
+    },
+    {
+      "link": [1, 4],
+      "name": "Come In Getting Blue Speed, Tricky Dash Jump",
+      "entranceCondition": {
+        "comeInGettingBlueSpeed": {
+          "length": 8,
+          "openEnd": 1
+        }
+      },
+      "requires": [
+        "canTrickyDashJump",
+        {"or": [
+          "canDodgeWhileShooting",
+          {"enemyDamage": {
+            "enemy": "Yellow Space Pirate (standing)",
+            "type": "contact",
+            "hits": 1
+          }}
+        ]},
+        {"heatFrames": 210}
+      ],
+      "note": [
+        "Come in with blue speed, run through the lowest Pirate, continuing with a short jump to kill the one on the right wall.",
+        "Next run a specific distance to get speed to jump onto the platform.",
+        "Then either quickly jump through the door before the top Pirate fires a laser, or wait for it to jump across the room and then jump over it."
+      ]
+    },
+    {
+      "link": [1, 4],
+      "name": "Come in Shinecharging, Leave Shinecharged (Pirate Dodge Wall Climb)",
+      "entranceCondition": {
+        "comeInShinecharging": {
+          "length": 8,
+          "openEnd": 1
+        }
+      },
+      "requires": [
+        "canShinechargeMovementComplex",
+        "canWalljump",
+        "canTrickyDodgeEnemies",
+        {"heatFrames": 145}
```

Comment:
I think in the video I just turn around while crouched, and then keep holding left to stand and walk for a few frames before spin-jumping? https://videos.maprando.com/video/1764

If I understand, what you're describing is press right to stand before turning around left and buffering a spin jump? I don't think I tried that. If that can work, sounds faster but more difficult to avoid the right Pirate. It might need low run speed so that you don't slide as far while crouching? In any case, is the "40" number seeming ok in the end, or were you wanting it changed?